CosmoTech_Acceleration_Library/Accelerators/scenario_download/scenario_downloader.py

```diff
@@ -274,29 +274,39 @@ def get_all_datasets(self, scenario_id: str) -> dict:
                 dataset_id = parameter['value']
                 dataset_ids.append(dataset_id)
 
-        def process(_dataset_id, _return_dict):
-            _c = self.download_dataset(_dataset_id)
-            if _dataset_id in self.dataset_file_temp_path:
-                _return_dict[_dataset_id] = (_c, self.dataset_file_temp_path[_dataset_id], _dataset_id)
-            else:
-                _return_dict[_dataset_id] = _c
+        def download_dataset_process(_dataset_id, _return_dict, _error_dict):
+            try:
+                _c = self.download_dataset(_dataset_id)
+                if _dataset_id in self.dataset_file_temp_path:
+                    _return_dict[_dataset_id] = (_c, self.dataset_file_temp_path[_dataset_id], _dataset_id)
+                else:
+                    _return_dict[_dataset_id] = _c
+            except Exception as e:
+                _error_dict[_dataset_id] = str(e)
```

Comment:
`str(e)` may not provide enough context to understand the problem in some cases (e.g. for `KeyError` it will only contain the key that was not found).
Maybe we could concatenate this with the error type? (e.g. `f'{type(e).__name__}: {str(e)}'`)

Reply:
That's better indeed.